libconcurrent/includes/sim.h

```diff
@@ -42,10 +42,10 @@ typedef struct SimObjectState {
 
 typedef union pointer_t {
     struct StructData {
-        int64_t seq : 32;
-        int32_t index : 32;
+        uint64_t seq : 40;
+        uint32_t index : 24;
```

Comment:
Minor: why not using `uint64_t index : 24;`? Too me looks more natural.

Reply:
I prefer to have more bits for sequence number.

Reply:
I meant that you use uint64_t for both member. Minor so it's ok as is.